src/lib/hooks/useBlockNumber.tsx

```diff
@@ -63,6 +71,12 @@ export function BlockNumberProvider({ children }: { children: ReactNode }) {
           console.error(`Failed to get block number for chainId ${activeChainId}`, error)
         })
 
+      if (activeChainId === ChainId.MAINNET) {
+        setEthBlockNumber(block)
```

Comment:
This has no guard against backslipping. Could you instead incorporate this into `onBlock`, something like

```
if (chainBlock.chainId === ChainId.MAINNET) {
  ...
}
```